src/algorithm/model.hxx

```diff
@@ -0,0 +1,211 @@
+          ModelTpl<Scalar,Options,JointCollectionTpl> &,
+          GeometryModel& > ArgsType;
+
+      template <typename JointModel>
+      static void algo (const JointModelBase<JointModel> & jmodel,
+          const ModelTpl<Scalar,Options,JointCollectionTpl> & modelAB,
+          const GeometryModel& geomModelAB,
+          JointIndex parentId,
+          SE3 pMi,
+          ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+          GeometryModel& geomModel)
+      {
+        // If old parent is universe, use what's provided in the input.
+        // otherwise, get the parent from modelAB.
+        if (modelAB.parents[jmodel.id()] > 0)
+          parentId = model.getJointId(modelAB.names[modelAB.parents[jmodel.id()]]);
+        JointIndex jid = model.addJoint (
+            parentId,
+            jmodel,
+            pMi * modelAB.jointPlacements[jmodel.id()],
+            modelAB.names[jmodel.id()],
+            jmodel.jointVelocitySelector(modelAB.effortLimit),
+            jmodel.jointVelocitySelector(modelAB.velocityLimit),
+            jmodel.jointConfigSelector(modelAB.lowerPositionLimit),
+            jmodel.jointConfigSelector(modelAB.upperPositionLimit));
+        assert (jid < model.joints.size());
+
+        model.appendBodyToJoint (jid, modelAB.inertias[jmodel.id()]);
+        // Add all frames whose parent is this joint.
+        for (FrameIndex fid = 1; fid < modelAB.frames.size(); ++fid) {
+          Frame frame = modelAB.frames[fid];
+          if (frame.parent == jmodel.id()) {
+            frame.parent = jid;
+            assert (frame.previousFrame > 0 || frame.type == JOINT);
+            if (frame.previousFrame != 0) {
+              frame.previousFrame = model.getFrameId (
+                  modelAB.frames[frame.previousFrame].name,
+                  modelAB.frames[frame.previousFrame].type);
+            }
+            model.addFrame (frame);
+          }
+        }
+        // Add all geometries whose parent is this joint.
+        for (GeomIndex gid = 0; gid < geomModelAB.geometryObjects.size(); ++gid) {
+          GeometryObject go = geomModelAB.geometryObjects[gid];
+          if (go.parentJoint == jmodel.id()) {
+            go.parentJoint = jid;
+            assert (go.parentFrame > 0);
+            if (go.parentFrame != 0) {
+              go.parentFrame = model.getFrameId (
+                  modelAB.frames[go.parentFrame].name,
+                  modelAB.frames[go.parentFrame].type);
+            }
+            geomModel.addGeometryObject (go);
+          }
+        }
+      }
+    };
+  }
+
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+  void
+  appendModel(const ModelTpl<Scalar,Options,JointCollectionTpl> & modelA,
+              const ModelTpl<Scalar,Options,JointCollectionTpl> & modelB,
+              const GeometryModel& geomModelA,
+              const GeometryModel& geomModelB,
+              FrameIndex frameInModelA,
+              SE3 aMb,
```

Comment:
I would put this as optional argument, defaulting to identity

Reply:
I was about to do that but I changed my mind. This function is not going to be used a lot and I prefer to have the outputs as last arguments.

Reply:
Yes, it sound logical to have the outputs at the end

Reply:
I mark this as resolved. @gabrielebndn unresolve if you disagree.

Reply:
Nope, agreed :)